### ISSUE TITLE
"invalid data for ascii" on rpc call for getrawtransaction

### ISSUE THREAD

**Issue by rllola** [namecoin/namecoin-core#337]
**Expected behavior**

Doing a RPC call with the following body 
```
{"jsonrpc": "1.0", "id":"curltest", "method": "getrawtransaction", "params": ["6b3b1f99f2ae9f9ee8e1420f0a7b60886575bf8e7c6144843d8ce07ebb00cbe5", true]}
```

We expect to have in nameOp the correct value.

**Actual behavior**

Instead we have an encoding error. 
```
            {
                "value": 0.01000000,
                "n": 1,
                "scriptPubKey": {
                    "nameOp": {
                        "op": "name_update",
                        "name": "d/zeronetwork",
                        "name_encoding": "ascii",
                        "value_error": "invalid data for ascii",
                        "value_encoding": "ascii"
                    },
                    "asm": "OP_NAME_UPDATE 642f7a65726f6e6574776f726b 7b226e73223a205b226e73312e646f6d61696e636f696e2e6e6574222c20226e73322e646f6d61696e636f696e2e6e6574225d2c200d0a227a65726f6e6574223a207b0d0a2022626c6f67223a202231424c6f6743394c4e346f50446372754e7a33716f31797361313333453941476738222c0d0a202274616c6b223a20223154614c6b337a4d375a52736b4a767268335a4e4344564758766b4a7573504b51222c0d0a2022223a202231455531746247396f433141386a7a326f755677475a7951356173724e7345345672220d0a7d0d0a7d OP_2DROP OP_DROP OP_DUP OP_HASH160 7308a8ae57775b8c9f278d6650414bc07a4f3317 OP_EQUALVERIFY OP_CHECKSIG",
                    "hex": "530d642f7a65726f6e6574776f726b4cd37b226e73223a205b226e73312e646f6d61696e636f696e2e6e6574222c20226e73322e646f6d61696e636f696e2e6e6574225d2c200d0a227a65726f6e6574223a207b0d0a2022626c6f67223a202231424c6f6743394c4e346f50446372754e7a33716f31797361313333453941476738222c0d0a202274616c6b223a20223154614c6b337a4d375a52736b4a767268335a4e4344564758766b4a7573504b51222c0d0a2022223a202231455531746247396f433141386a7a326f755677475a7951356173724e7345345672220d0a7d0d0a7d6d7576a9147308a8ae57775b8c9f278d6650414bc07a4f331788ac",
                    "reqSigs": 1,
                    "type": "pubkeyhash",
                    "addresses": [
                        "N74cHLeJe4NpRV8q7WydMbuz77a8hgRosq"
                    ]
                }
```

**To reproduce**

1. Start a namecoin node with rpc server feature enable
2. get raw transaction example `{"jsonrpc": "1.0", "id":"curltest", "method": "getrawtransaction", "params": ["6b3b1f99f2ae9f9ee8e1420f0a7b60886575bf8e7c6144843d8ce07ebb00cbe5", true]}`
3. Get the encoding error

** IMPORTANT **

It doesn't seem to be reproducible for more recent transaction.

**Comment by domob1812** [comment]
The issue is that the name's value presumably contains non-ASCII characters (I can't check it right now).  To see them, you need to choose a different value encoding.  This can either be done with `-valueencoding` on the commandline or with `valueending=` in your `namecoin.conf`.  (Other RPC methods like `name_show` allow you to also choose the encoding with an RPC argument, but I think `getrawtransaction` doesn't allow that.)

**Comment by rllola** [comment]
Thank you @domob1812 
I will try changing `valueending`.

I just want to mentioned that it used to work with precedent version of namecoin core. Not sure why it does that for namecoin core 0.18.0 

Here the transaction and it shows the expected value :
https://namecha.in/tx/6b3b1f99f2ae9f9ee8e1420f0a7b60886575bf8e7c6144843d8ce07ebb00cbe5

**Comment by domob1812** [comment]
This was changed in #249.  Previously, we simply copied the name/value data to a JSON string, without checking it in any way.  This worked for a lot of cases (e.g. UTF-8), but also broke RPC clients with invalid JSON data for other cases (e.g. binary data or invalid UTF-8).  With the change, we now fully sanitise the data, and allow clients to choose what kind of data they want to receive and can handle.

In your case, it looks like the issue is just that the name value JSON includes whitespace and in particular newline characters.  So the proper solution is to stop wasting blockchain space and tx fees and just using a compact, whitespace-less, serialisation of the JSON values.

In other cases where the value is genuinely binary data, the `hex` encoding can now be used to properly handle this data through the RPC interface, which was not possible before.
